recipe/thirdparty/api/utils.go

```diff
@@ -1,48 +1,168 @@
 package api
 
-import "github.com/supertokens/supertokens-golang/recipe/thirdparty/tpmodels"
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"sync"
 
-func findRightProvider(providers []tpmodels.TypeProvider, thirdPartyId string, clientId *string) *tpmodels.TypeProvider {
+	"github.com/derekstavis/go-qs"
+	"github.com/supertokens/supertokens-golang/recipe/thirdparty/providers"
+	"github.com/supertokens/supertokens-golang/recipe/thirdparty/tpmodels"
+	"github.com/supertokens/supertokens-golang/supertokens"
+)
 
-	/* logic to find the right provider
+func findProvider(options tpmodels.APIOptions, thirdPartyId string, tenantId string) (tpmodels.TypeProvider, error) {
 
-	ClientID not passed
-		Case 1: Single Config - Return the config
-		Case 2: Multiple Config - Return the config with IsDefault = true
+	for _, provider := range options.Providers {
+		if provider.ID == thirdPartyId {
+			return provider, nil
```

Comment:
change to check if !provider.useForDefault && tenantId == default -> return error